packages/azure/src/index.ts

```
@@ -49,3 +51,22 @@ export function cosmosQueryClient(configuration: ICosmosConfiguration): ICosmosQ
     configuration = config.parse(CosmosConfiguration, configuration);
     return new CosmosClient(configuration);
 }
+
+export interface IBlobClient extends IRequireInitialization {
+    createContainerIfNotExists(context?: SpanContext): Promise<BlobService.ContainerResult>;
+
+    writeAsText(context: SpanContext, text: Buffer | string, blobId: string): Promise<void>;
+    readAsText(context: SpanContext, blobId: string): Promise<string>;
+    writeAsLargeText(text: string, blobId: string, context: SpanContext): Promise<void>;
```

what's the difference between `writeAsLargeText` and `writeAsText`? why do I have to choose as an end-user? Both accept a string to be written

---

Oh I see. I can just expose writeAsText() and depending on the size of the string, I can handle things differently.

---

one downside to having just a single method for writing text however is that the clients will always need to provide the localStoragePath parameter for IBlobStorageConfiguration. 

So maybe we should move writeAsLargeText() into its own class LargeBlobClient, then there wont any confusion.

---

I would suggest the following, rename the method to just `write(context: SpanContext, blobId: string, value: Buffer | string)` (you also accept buffers, so it isn't just text). And then internally if the localStoragePath is set AND either the buffer or the string is longer than X, then do the optimized upload version (I assume it would also work with binary data and not just text if there is a larger Buffer). Maybe we can also find a better name for `localStoragePath` that describes its purpose better ... maybe `largeContentTempPath` or something like that?

---

I can check if the blob library also accepts large buffers. 

and if the user forgets to set the storage path, then we could throw an exception